addons/event-publisher/common/src/main/java/org/commonjava/indy/event/publisher/KafkaEventPublisher.java

```diff
@@ -0,0 +1,141 @@
+package org.commonjava.indy.event.publisher;
+
+import org.commonjava.event.file.FileEvent;
+import org.commonjava.event.file.FileEventType;
+import org.commonjava.indy.IndyWorkflowException;
+import org.commonjava.indy.conf.IndyConfiguration;
+import org.commonjava.indy.content.ContentDigester;
+import org.commonjava.indy.core.conf.IndyEventHandlerConfig;
+import org.commonjava.indy.data.IndyDataException;
+import org.commonjava.indy.data.StoreDataManager;
+import org.commonjava.indy.folo.model.TrackingKey;
+import org.commonjava.indy.model.core.RemoteRepository;
+import org.commonjava.indy.model.core.StoreKey;
+import org.commonjava.indy.model.core.StoreType;
+import org.commonjava.indy.model.galley.KeyedLocation;
+import org.commonjava.maven.galley.event.EventMetadata;
+import org.commonjava.maven.galley.event.FileAccessEvent;
+import org.commonjava.maven.galley.event.FileDeletionEvent;
+import org.commonjava.maven.galley.event.FileStorageEvent;
+import org.commonjava.maven.galley.io.checksum.ContentDigest;
+import org.commonjava.maven.galley.io.checksum.TransferMetadata;
+import org.commonjava.maven.galley.model.Location;
+import org.commonjava.maven.galley.model.Transfer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+import java.util.Date;
+
+public class KafkaEventPublisher
+                implements FileEventPublisher
+{
+
+    private final Logger logger = LoggerFactory.getLogger( getClass() );
+
+    @Inject
+    private IndyEventHandlerConfig handlerConfig;
+
+    @Inject
+    private IndyConfiguration indyConfig;
+
+    @Inject
+    StoreDataManager storeManager;
+
+    @Inject
+    ContentDigester contentDigester;
+
+    public void onFileDelete( @Observes final FileDeletionEvent event )
+    {
+
+        if ( !IndyEventHandlerConfig.HANDLER_KAFKA.equals( handlerConfig.getFileEventHandler() ))
+        {
+            return;
+        }
+        FileEvent fileEvent = new FileEvent( FileEventType.DELETE );
+        transformFileEvent( event, fileEvent );
+        publishFileEvent( fileEvent );
+
+    }
+
+    public void onFileUpload( @Observes final FileStorageEvent event )
+    {
+
+        if ( !IndyEventHandlerConfig.HANDLER_KAFKA.equals( handlerConfig.getFileEventHandler() ))
+        {
+            return;
+        }
+
+        FileEvent fileEvent = new FileEvent( FileEventType.STORAGE );
+        transformFileEvent( event, fileEvent );
+        publishFileEvent( fileEvent );
+    }
+
+    private void transformFileEvent( org.commonjava.maven.galley.event.FileEvent event, FileEvent fileEvent )
+    {
+        EventMetadata metadata = event.getEventMetadata();
+        final TrackingKey trackingKey = (TrackingKey) metadata.get( "tracking-id" );
+        if ( trackingKey == null )
```

Comment:
Does this mean we only focus some folo cases here?

Reply:
Hm, this checking should be removed, and the publisher should not do any special checking. I will update it.